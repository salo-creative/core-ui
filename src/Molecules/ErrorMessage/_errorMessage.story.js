import React from 'react';
import { storiesOf } from '@storybook/react';

// load tests
import { withTests } from '@storybook/addon-jest';
import { withKnobs, text } from '@storybook/addon-knobs';
import results from '../../../.storybook/jest-test-results.json';

// FEATURED COMPONENTS //
import { ErrorMessage } from '../../index';

// README //
import README from './README.md';

// Start of story logic
const stories = storiesOf('Molecules | ErrorMessage', module);
stories.addDecorator(withKnobs);
stories.addDecorator(withTests({ results }));
stories.addParameters({ jest: ['errormessage'] });

stories.add(
  'Basic',
  (() => {
    // KNOBS
    const title = text('Title', 'Error title');
    return (
      <ErrorMessage
        error={ {} }
        title={ title }
        retryAction={ () => alert('Retry this') }
      />
    );
  }), { notes: README }
);