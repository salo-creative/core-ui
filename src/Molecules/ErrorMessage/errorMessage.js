import React from 'react';
import PropTypes from 'prop-types';

// COMPONENTS & STYLES
import H3 from '../../Typography/H3';
import P from '../../Typography/P';
import Button from '../Button';
import ApolloError from '../../Apollo/Error';
import { ErrorContainer } from './errorMessage.styles';

const ErrorMessage = (props) => {
  const {
    error,
    padding,
    retryAction,
    title
  } = props;

  return (
    <ApolloError
      error={ error }
      addAlert={ false }
    >
      { ({ message, code }) => (
        <ErrorContainer width='700px' padding={ padding }>
          <H3 align='center' margin='0 0 1rem'>{ title }</H3>
          <P align='center'>Status: { code || 400 } | { message || 'Sorry the action you have performed is invalid. Please check the details supplied and try again' }</P>
          { retryAction && typeof retryAction === 'function' && (
            <React.Fragment>
              <P align='center'>You can retry the last action by pressing the button below.</P>
              <Button onClick={ retryAction } iconBefore='sync' />
            </React.Fragment>
          ) }
        </ErrorContainer>
      ) }
    </ApolloError>
  );
};

ErrorMessage.defaultProps = {
  padding: '2rem',
  retryAction: null,
  title: 'Whoops something went wrong'
};

ErrorMessage.propTypes = {
  error: PropTypes.object.isRequired,
  padding: PropTypes.string,
  retryAction: PropTypes.func,
  title: PropTypes.string
};

export default ErrorMessage;