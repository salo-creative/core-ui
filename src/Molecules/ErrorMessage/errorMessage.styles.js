import styled from 'styled-components';
import Container from '../../Atoms/Container';

export const ErrorContainer = styled(Container)`
  align-items: center;
  display: flex;
  flex-direction: column;
  justify-content: center;
  padding: ${ ({ padding }) => padding }
`;