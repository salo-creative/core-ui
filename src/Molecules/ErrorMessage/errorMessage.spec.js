import React from 'react';
import 'jest-styled-components';

import { renderWithTheme } from '../../../test';

// COMPONENT
import ErrorMessage from './index';

test('Renders against snapshot', async () => {
  const { container } = renderWithTheme(
    <ErrorMessage error={ {
      graphQLErrors: [{
        message: 'SC_ERROR'
      }]
    } }
    />
  );
  expect(container.firstChild).toMatchSnapshot();
});