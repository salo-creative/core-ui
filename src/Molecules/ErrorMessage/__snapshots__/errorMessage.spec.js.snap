// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders against snapshot 1`] = `
.c1 {
  width: 100%;
  color: #333333;
  text-align: center;
  margin: 0 0 1rem;
  padding: 0;
  font-weight: 700;
  font-size: 17px;
  line-height: 27px;
}

.c2 {
  font-weight: 400;
  width: 100%;
  color: #333333;
  text-align: center;
  margin: 0 0 20px;
  padding: 0;
  font-size: 14px;
  line-height: 17px;
}

.c2 em,
.c2 i {
  font-style: italic;
}

.c2 b,
.c2 strong {
  font-weight: bold;
}

.c2 a {
  font-size: 1em;
  color: currentColor;
  -webkit-transition: color 0.3s;
  transition: color 0.3s;
  font-weight: 600;
}

.c2 a:hover {
  -webkit-transition: color 0.3s;
  transition: color 0.3s;
  color: #8f8f8f;
}

.c2 pre {
  font-family: "Courier 10 Pitch",Courier,monospace;
  font-size: 95%;
  line-height: 140%;
  white-space: pre;
  white-space: -moz-pre-wrap;
  white-space: -o-pre-wrap;
  white-space: pre-wrap;
}

.c2 code {
  font-family: Monaco,Consolas,"Andale Mono","DejaVu Sans Mono",monospace;
  font-size: 95%;
  line-height: 140%;
  white-space: pre;
  white-space: -moz-pre-wrap;
  white-space: -o-pre-wrap;
  white-space: pre-wrap;
  background: #faf8f0;
}

.c0 {
  width: 100%;
  margin: 0 auto;
  position: relative;
  max-width: 700px;
  padding: 2rem;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 2rem;
}

@media screen and (min-width:400px) {
  .c1 {
    font-size: calc(17px + (27 - 17) * (100vw - 400px) / (1200 - 400));
    line-height: calc(27px + (34 - 27) * (100vw - 400px) / (1200 - 400));
  }
}

@media screen and (min-width:1200px) {
  .c1 {
    font-size: 27px;
    line-height: 34px;
  }
}

@media screen and (min-width:400px) {
  .c2 {
    font-size: calc(14px + (17 - 14) * (100vw - 400px) / (1200 - 400));
    line-height: calc(17px + (29 - 17) * (100vw - 400px) / (1200 - 400));
  }
}

@media screen and (min-width:1200px) {
  .c2 {
    font-size: 17px;
    line-height: 29px;
  }
}

<div
  class="c0"
  width="700px"
>
  <h3
    class="c1"
    color=""
    font-size=""
  >
    Whoops something went wrong
  </h3>
  <p
    class="c2"
    color=""
    font-size=""
  >
    Status: 
    400
     | 
    Something went wrong!
  </p>
</div>
`;
